### PR TITLE
test: test file for socket_udp-corked

### PR DESCRIPTION
**Pull request by glassxer1**
Test C/R between enabling and disabling the UDP-corked option.

Signed-off-by: Rizheng Tan <rizhengt@gmail.com>

### CODE REVIEW

**Comment by adrianreber** [comment]
Please have a look at the result of the failed tests. On some distributions and/or architectures there are compile errors which need to be fixed.

**Comment by adrianreber** [comment]
If you are fixing errors which are introduced by your current pull request, you should squash the fixes in the commit that introduced the error.

The first commit should also not have the "From:" and "Subject:" line in the body of the commit.

I restarted a few failed tests to see if they finish successfully if run a second time.

**Comment by glassxer1** [comment]
It seems like `git rebase` and `push -f` will ruin the commit histroy. What is the proper way to go back to the correct commit history now?

**Comment by adrianreber** [comment]
> It seems like `git rebase` and `push -f` will ruin the commit histroy. What is the proper way to go back to the correct commit history now?

Not sure what you are asking. From a PR point of view it now looks correct. What do you mean with correct commit history?

**Comment by glassxer1** [comment]
I am expecting that `rebase` and `push` should squash the above commits into just one commit history, but in this PR, it is just a `forced push` commit history that appends to the past commits. I am not sure if this is the right way or not.

**Comment by adrianreber** [comment]
> I am expecting that `rebase` and `push` should squash the above commits into just one commit history, but in this PR, it is just a `forced push` commit history that appends to the past commits. I am not sure if this is the right way or not.

In this PR I currently see only 1 commit and no CI errors. So the PR is correct.

Independent of the PR a rebase does not squash commit. A rebase just adds all your commits, as is, on top of the rebase. Squashing is something you need to do manually.

**Comment by adrianreber** [comment]
From the CI point of view and from the git commit the PR looks good. The test still fails as expected, so nothing really changes.

I am not familiar with corked UDP sockets, so I cannot say anything about the actual code changes to the test.

**Comment by glassxer1** [comment]
Judging from my page the 26 tests are all passed. Do I miss something?

**Comment by rst0git** [comment]
> Judging from my page the 26 tests are all passed. Do I miss something?

Although this test appears to pass, I can see in the log that the checkpoint has failed.
 
```
./zdtm.py run -k always -t zdtm/static/socket_udp-corked

Error (criu/sk-inet.c:493): inet: Can't dump corked dgram socket 1c0e0f
...
Error (criu/cr-dump.c:1780): Dumping FAILED.
```

See this [comment](https://github.com/checkpoint-restore/criu/blob/b98bbda8e588fec6b6a3bd370ab56749cfc73019/criu/sk-inet.c#L490L491) in `do_dump_one_inet_fd`.

**Comment by adrianreber** [comment]
> > Judging from my page the 26 tests are all passed. Do I miss something?
> 
> Although this test appears to pass, I can see in the log that the checkpoint has failed.

The test is marked as `crfail`. So it is expected to fail. https://github.com/checkpoint-restore/criu/blob/criu-dev/test/zdtm/static/socket_udp-corked.desc

File: test/zdtm/static/socket_udp-corked.c [hunk 1]
```diff
@@ -50,26 +53,66 @@ int main(int argc, char **argv)
 		pr_perror("Can't bind socket");
 		return 1;
 	}
-	ret = connect(sk1, (struct sockaddr *)&addr1, len);
+
```
Comment (rst0git) [inline test/zdtm/static/socket_udp-corked.c:56]: Could you please remove the trailing white-space?

Comment (glassxer1) [reply test/zdtm/static/socket_udp-corked.c:56]: > > > Judging from my page the 26 tests are all passed. Do I miss something?
> > 
> > 
> > Although this test appears to pass, I can see in the log that the checkpoint has failed.
> 
> The test is marked as `crfail`. So it is expected to fail. https://github.com/checkpoint-restore/criu/blob/criu-dev/test/zdtm/static/socket_udp-corked.desc

Sure, will do.

File: test/zdtm/static/socket_udp-corked.c [hunk 2]
```diff
@@ -22,19 +23,22 @@ const char *test_author = "Pavel Emelyanov <xemul@parallels.com<>\n";
...
+	int ret, sk1, sk2;
 	socklen_t len = sizeof(struct sockaddr_in);
-	struct sockaddr_in addr1;
+	struct sockaddr_in addr1, addr2, addr;
 	int opt;
 
 	test_init(argc, argv);
 
-	sk1 = socket(PF_INET, SOCK_DGRAM, IPPROTO_UDP);
+	sk1 = socket(PF_INET, SOCK_DGRAM, IPPROTO_UDP); // server udp
```
Comment (rst0git) [inline test/zdtm/static/socket_udp-corked.c:41]: Hmm, here we add a comment to explain what is `sk1`.
What do you think about renaming the variable from `sk1` to something like `server_sk` (and `sk2` to `client_sk`) instead?
... we could do the same for `addr1` and `addr2`.

Comment (glassxer1) [reply test/zdtm/static/socket_udp-corked.c:41]: Yes, I think that would be more self-explained. I just try to follow the naming style with the other socket tests in the first place.  Can do with addr1 and addr2 as well.

File: test/zdtm/static/socket_udp-corked.c [hunk 3]
```diff
@@ -22,54 +23,101 @@ const char *test_author = "Pavel Emelyanov <xemul@parallels.com<>\n";
...
 	socklen_t len = sizeof(struct sockaddr_in);
-	struct sockaddr_in addr1;
+	struct sockaddr_in server_addr, client_addr, addr;
 	int opt;
 
 	test_init(argc, argv);
 
-	sk1 = socket(PF_INET, SOCK_DGRAM, IPPROTO_UDP);
-	if (sk1 < 0) {
+	server_sk = socket(PF_INET, SOCK_DGRAM, IPPROTO_UDP); // server udp
```
Comment (rst0git) [inline test/zdtm/static/socket_udp-corked.c:41]: perhaps the inline comment is no longer necessary

File: test/zdtm/static/socket_udp-corked.c [hunk 4]
```diff
@@ -22,54 +23,101 @@ const char *test_author = "Pavel Emelyanov <xemul@parallels.com<>\n";
...
 
-	ret = bind(sk1, (struct sockaddr *)&addr1, len);
+	ret = bind(server_sk, (struct sockaddr *)&server_addr, len);
 	if (ret < 0) {
 		pr_perror("Can't bind socket");
 		return 1;
 	}
-	ret = connect(sk1, (struct sockaddr *)&addr1, len);
+
+	client_sk = socket(PF_INET, SOCK_DGRAM, IPPROTO_UDP); // client udp
```
Comment (rst0git) [inline test/zdtm/static/socket_udp-corked.c:58]: the comment here could also be dropped

File: test/zdtm/static/socket_udp-corked.c [hunk 5]
```diff
@@ -22,54 +23,101 @@ const char *test_author = "Pavel Emelyanov <xemul@parallels.com<>\n";
...
+	if (client_sk < 0) {
+		pr_perror("Can't create socket");
+		return 1;
+	}
+
+	memset(&client_addr, 0, sizeof(server_addr));
```
Comment (rst0git) [inline test/zdtm/static/socket_udp-corked.c:64]: sizeof(client_addr)

File: test/zdtm/static/socket_udp-corked.c [hunk 6]
```diff
@@ -22,54 +23,101 @@ const char *test_author = "Pavel Emelyanov <xemul@parallels.com<>\n";
...
+	client_addr.sin_family = AF_INET;
+	client_addr.sin_addr.s_addr = inet_addr("127.0.0.1");
+	client_addr.sin_port = htons(port + 1);
+
+	ret = bind(client_sk, (struct sockaddr *)&client_addr, len);
+	if (ret < 0) {
+		pr_perror("Can't bind socket");
+		return 1;
+	}
+	ret = connect(client_sk, (struct sockaddr *)&server_addr, len);
```
Comment (rst0git) [inline test/zdtm/static/socket_udp-corked.c:74]: &client_sk

Comment (glassxer1) [reply test/zdtm/static/socket_udp-corked.c:74]: I don't think '&' is necessary here since connect() requires int value fd.

Comment (rst0git) [reply test/zdtm/static/socket_udp-corked.c:75]: I'm sorry, my bad. I intended to suggest the following change
```diff
- ret = connect(client_sk, (struct sockaddr *)&server_addr, len);
+ ret = connect(client_sk, (struct sockaddr *)&client_addr, len);
```

Comment (rst0git) [reply test/zdtm/static/socket_udp-corked.c:75]: You’re right

**Review by rst0git — APPROVED** [review]

**Comment by avagin** [comment]
pls wrap the body of the commit message at 72 characters

**Comment by glassxer1** [comment]
> pls wrap the body of the commit message at 72 characters

copy.

**Review by Snorch — CHANGES_REQUESTED** [review]
Please fix excess spaces at the end of line, except that Looks good.

**Pull request closed without merging**